### DISCUSSION TITLE
Kibana borked post upgrade

### DISCUSSION THREAD

**Discussion by WPTechnician** [Security-Onion-Solutions/securityonion#3280] [Unsupported Versions] [answered]
Hi:

I just upgraded to 2.3.30 from 2.3.21 and it seemed to go fine.  After it was complete, I found that Kibana was reporting:

{"type":"log","@timestamp":"2021-03-04T14:25:22Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Another Kibana instance appears to be migrating the index. Waiting for that migration to complete. If no other Kibana instance is attempting migrations, you can get past this message by deleting index .kibana_2 and restarting Kibana."}

I deleted the .kibana_2 index and restarted and then ran the so-kibana-config-load command and kibana was now accessible but all of the dashboards are gone.  I can see 16.04 dashboards but nothing else.  Is there anything else I need to do with this to recover?

Thanks!

**Comment by WPTechnician** [comment, marked as answer]
So it appears the dashboards are there, they just don't come up by default anymore, they have to be browsed to.

**Reply by WPTechnician** [reply]
Ok, I have that sorted as well.  Just have to update the default view in the Stack Management to the Security Onion Home.  All good now!

**Reply by Cingrith** [reply]
how do you delete the .kibana_2 index? I'm having a similar issue, but I'm still using 16.0.4.

**Reply by Cingrith** [reply]
never mind I got it.......... one of those days

**Reply by walpman** [reply]
Hi there, I'm a newbie, but am having the exact same issue, not a DBA either, how and where can I find/delete the .kibana_2 index? I assume it's inside a MySQL? thanks ahead of time

**Reply by WPTechnician** [reply]
> Hi there, I'm a newbie, but am having the exact same issue, not a DBA either, how and where can I find/delete the .kibana_2 index? I assume it's inside a MySQL? thanks ahead of time

Sure,  curl -X DELETE "localhost:9200/.kibana_2"

**Reply by walpman** [reply]
thank you, looks like it worked, I have to browse to the dashboards, but at least my data is loading.

**Reply by WPTechnician** [reply]
So, to fix that:

1. Open Kibana
2. Browse to the Security Onion Home dashboard
3. Copy everything in the address bar after kibana including the forward slash (/app/dashboards#/view/...etc) right to the end
4. Click the hamburger at the top left of the screen
5. Click Stack Management
6. Click Advanced Settings
7. Find the Default Route setting
8. Replace the current setting with what you copied above
9. Click the save settings button at the bottom

If you did it right the next time you click the Kibana link from SO it will take you right to the home dashboard.

**Reply by DDD-Digithead** [reply]
Same exact problem here now except "1. Open Kibana" isn't possible, returns a 404 error.
How do I get around that?

**Reply by WPTechnician** [reply]
Just to check, you ran the update and after that Kibana was reporting an issue so you deleted the index and now it is reporting not found? What are the logs saying?  If you run so-status on the manager what do you see?  More info here:
https://docs.securityonion.net/en/2.3/kibana.html

**Reply by DDD-Digithead** [reply]
Negative. Deleted, changed, modified nothing yet.

**Reply by walpman** [reply]
Thanks @WPTechnician but I did I so-kibana stop and start and the link from the SO dashboard works now.

**Reply by DDD-Digithead** [reply]
so-status is showing that everything is running. However Grafana shows 0 activity for Kibana even after so-kibana-stop / so-kibana-start. Looking for the log files right now. You'll have to forgive me as I'm a total neophyte with this as I'm using it on the home network as a learning tool.

**Reply by WPTechnician** [reply]
No worries!  I haven't seen that error in relation to Kibana.  If it is running but not answering on the port something is pretty broken.  At least if it is a test instance reinstalling can get you back in business relatively quickly.

**Reply by DDD-Digithead** [reply]
Actually, would be a bit more exciting to find the problem and correct it. But not sure that's within my wheel-house. The logs for Kibana seem to indicate there is a rather serious problem though. Am also experiencing issues with Alert and acknowledgements. It tends to cause that to stop responding as well. But here is what I'm seeing in the Kibana logs:
{"type":"log","@timestamp":"2021-03-05T13:50:54Z","tags":["info","plugins-service"],"pid":293,"message":"Plugin \"telemetryManagementSection\" has been disabled since the following direct or transitive dependencies are missing or disabled: [telemetry]"}
{"type":"log","@timestamp":"2021-03-05T13:50:54Z","tags":["info","plugins-service"],"pid":293,"message":"Plugin \"telemetry\" is disabled."}
{"type":"log","@timestamp":"2021-03-05T13:50:54Z","tags":["info","plugins-service"],"pid":293,"message":"Plugin \"visTypeXy\" is disabled."}
{"type":"log","@timestamp":"2021-03-05T13:50:54Z","tags":["warning","config","deprecation"],"pid":293,"message":"\"cpu.cgroup.path.override\" is deprecated and has been replaced by \"ops.cGroupOverrides.cpuPath\""}
{"type":"log","@timestamp":"2021-03-05T13:50:54Z","tags":["warning","config","deprecation"],"pid":293,"message":"\"cpuacct.cgroup.path.override\" is deprecated and has been replaced by \"ops.cGroupOverrides.cpuAcctPath\""}
{"type":"log","@timestamp":"2021-03-05T13:50:54Z","tags":["warning","config","deprecation"],"pid":293,"message":"You should set server.basePath along with server.rewriteBasePath. Starting in 7.0, Kibana will expect that all requests start with server.basePath rather than expecting you to rewrite the requests in your reverse proxy. Set server.rewriteBasePath to false to preserve the current behavior and silence this warning."}
{"type":"log","@timestamp":"2021-03-05T13:50:55Z","tags":["info","plugins-system"],"pid":293,"message":"Setting up [39] plugins: [usageCollection,telemetryCollectionManager,kibanaUsageCollection,securityOss,newsfeed,mapsLegacy,kibanaLegacy,share,legacyExport,bfetch,embeddable,expressions,data,home,apmOss,console,management,indexPatternManagement,advancedSettings,savedObjects,dashboard,visualizations,visTypeVega,visTypeTable,visTypeTimelion,timelion,visTypeMarkdown,tileMap,regionMap,inputControlVis,visualize,esUiShared,charts,visTypeVislib,visTypeTagcloud,visTypeTimeseries,visTypeMetric,discover,savedObjectsManagement]"}
{"type":"log","@timestamp":"2021-03-05T13:50:56Z","tags":["info","savedobjects-service"],"pid":293,"message":"Waiting until all Elasticsearch nodes are compatible with Kibana before starting saved objects migrations..."}
{"type":"log","@timestamp":"2021-03-05T13:50:56Z","tags":["info","savedobjects-service"],"pid":293,"message":"Starting saved objects migrations"}
{"type":"log","@timestamp":"2021-03-05T13:50:56Z","tags":["info","savedobjects-service"],"pid":293,"message":"Creating index .kibana_2."}
{"type":"log","@timestamp":"2021-03-05T13:50:57Z","tags":["error","elasticsearch","data"],"pid":293,"message":"[resource_already_exists_exception]: index [.kibana_2/63RxbuFoRh2QvUYGeJI5tA] already exists"}
{"type":"log","@timestamp":"2021-03-05T13:50:57Z","tags":["warning","savedobjects-service"],"pid":293,"message":"Unable to connect to Elasticsearch. Error: resource_already_exists_exception"}
{"type":"log","@timestamp":"2021-03-05T13:50:57Z","tags":["warning","savedobjects-service"],"pid":293,"message":"Another Kibana instance appears to be migrating the index. Waiting for that migration to complete. If no other Kibana instance is attempting migrations, you can get past this message by deleting index .kibana_2 and restarting Kibana."}
{"type":"log","@timestamp":"2021-03-05T13:52:04Z","tags":["error","elasticsearch","data"],"pid":293,"message":"[circuit_breaking_exception]: [parent] Data too large, data for [<http_request>] would be [4035765040/3.7gb], which is larger than the limit of [4028419276/3.7gb], real usage: [4035765040/3.7gb], new bytes reserved: [0/0b], usages [request=1802240/1.7mb, fielddata=14106/13.7kb, in_flight_requests=0/0b, accounting=519050004/495mb]"}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["info","plugins-service"],"pid":292,"message":"Plugin \"telemetryManagementSection\" has been disabled since the following direct or transitive dependencies are missing or disabled: [telemetry]"}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["info","plugins-service"],"pid":292,"message":"Plugin \"telemetry\" is disabled."}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["info","plugins-service"],"pid":292,"message":"Plugin \"visTypeXy\" is disabled."}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["warning","config","deprecation"],"pid":292,"message":"\"cpu.cgroup.path.override\" is deprecated and has been replaced by \"ops.cGroupOverrides.cpuPath\""}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["warning","config","deprecation"],"pid":292,"message":"\"cpuacct.cgroup.path.override\" is deprecated and has been replaced by \"ops.cGroupOverrides.cpuAcctPath\""}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["warning","config","deprecation"],"pid":292,"message":"You should set server.basePath along with server.rewriteBasePath. Starting in 7.0, Kibana will expect that all requests start with server.basePath rather than expecting you to rewrite the requests in your reverse proxy. Set server.rewriteBasePath to false to preserve the current behavior and silence this warning."}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["info","plugins-system"],"pid":292,"message":"Setting up [39] plugins: [usageCollection,telemetryCollectionManager,kibanaUsageCollection,securityOss,newsfeed,mapsLegacy,kibanaLegacy,share,legacyExport,embeddable,expressions,data,home,console,apmOss,management,indexPatternManagement,advancedSettings,savedObjects,dashboard,visualizations,visTypeVega,visTypeTimelion,timelion,visTypeTable,visTypeMarkdown,tileMap,regionMap,inputControlVis,visualize,esUiShared,charts,visTypeVislib,visTypeTimeseries,visTypeTagcloud,visTypeMetric,discover,savedObjectsManagement,bfetch]"}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["info","savedobjects-service"],"pid":292,"message":"Waiting until all Elasticsearch nodes are compatible with Kibana before starting saved objects migrations..."}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["info","savedobjects-service"],"pid":292,"message":"Starting saved objects migrations"}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["info","savedobjects-service"],"pid":292,"message":"Creating index .kibana_2."}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["error","elasticsearch","data"],"pid":292,"message":"[resource_already_exists_exception]: index [.kibana_2/63RxbuFoRh2QvUYGeJI5tA] already exists"}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Unable to connect to Elasticsearch. Error: resource_already_exists_exception"}
{"type":"log","@timestamp":"2021-03-05T14:08:41Z","tags":["warning","savedobjects-service"],"pid":292,"message":"Another Kibana instance appears to be migrating the index. Waiting for that migration to complete. If no other Kibana instance is attempting migrations, you can get past this message by deleting index .kibana_2 and restarting Kibana."}
{"type":"log","@timestamp":"2021-03-05T14:11:10Z","tags":["error","elasticsearch","data"],"pid":292,"message":"[circuit_breaking_exception]: [parent] Data too large, data for [<http_request>] would be [4061910144/3.7gb], which is larger than the limit of [4028419276/3.7gb], real usage: [4061910144/3.7gb], new bytes reserved: [0/0b], usages [request=0/0b, fielddata=14106/13.7kb, in_flight_requests=344/344b, accounting=519405388/495.3mb]"}

**Reply by WPTechnician** [reply]
It appears that the index is still a problem.  Try:

1. sudo curl -X DELETE "localhost:9200/.kibana_2"
2. sudo so-kibana-restart
3. Give it a few minutes to restart
4. sudo so-kibana-config-load

After that, give it a try.

**Reply by DDD-Digithead** [reply]
>BOOM!< That did the trick! Didn't have much hope as I had already tried the first and second lines, but not the 4'th. Kibana is alive again. Thank you!   :-D

**Reply by WPTechnician** [reply]
Good stuff!  Have a good one!

**Reply by DDD-Digithead** [reply]
Indeed! I have a stupidly complex home network... but it's sure a lot of fun to tinker with and helps keep my skills sharp (work as an engineer for an MSP and trying to learn WAY more about security).
Thanks again! Deeply appreciated.

**Reply by WPTechnician** [reply]
Oof, I left my MSP days behind me in 2009.  I am a sysadmin at a hospital now and have become the defacto security lead.  It is a gigantic field, definitely keeping things interesting after 24 years in IT.

Glad to have helped, Security Onion is awesome!  We had been using a Trendmicro product and while it isn't so bad the renewal for it was like 60K.  I spun up an SO cluster on a couple of HPE DL380G8s and sent the same traffic and out of the box it is discovering the same and more compared to the DDI appliance. Just great!

**Reply by DDD-Digithead** [reply]
Hhmm... still having problems with it but will have to look into it tomorrow morning.

And yes, I have to agree. I had worked as a "One-Man-Band IT Shop" for 13 years before coming onboard where I am now. Starting to think that was a BIG mistake. Not real thrilled with the MSP arena now.

**Reply by DDD-Digithead** [reply]
Well this thing is toast. Going to have to just reload from scratch again I guess. Second time an update has broken the daylights out of it.

**Reply by sleepingbel** [reply]
@WPTechnician Thx your solutions works for me!!!

**Comment by dougburks** [comment]
We've updated our documentation:
https://docs.securityonion.net/en/2.3/soup.html#kibana
https://docs.securityonion.net/en/2.3/kibana.html#diagnostic-logging

**Reply by DDD-Digithead** [reply]
That saved my bacon this morning. Thank you.
